result/src/test/kotlin/com/github/kittinunf/result/ResultTests.kt

```diff
@@ -231,6 +231,18 @@ class ResultTests {
         assertThat("v2 is failure", v2.component2() is IllegalArgumentException, equalTo(true))
     }
 
+    @Test
+    fun onError() {
+        val success = Result.of("success")
+        val failure = Result.error(Exception("failure"))
+
+        val v1 = success.onError { }
```

Comment:
Would you mind test something a bit more meaningful inside the block? Like setting some value and check it whether it is being set?

Reply:
Sure :) Now this test checks if `var hasFailureChanged` changed during execution of the function.

```kotlin
@Test
fun onError() {
        val success = Result.of("success")
        val failure = Result.error(Exception("failure"))

        var hasSuccessChanged = false
        var hasFailureChanged = false

        val v1 = success.onError { hasSuccessChanged = true }
        val v2 = failure.onError { hasFailureChanged = true }

        assertThat("v1 is success", v1, instanceOf(Result.Success::class.java))
        assertThat("v2 is failure", v2, instanceOf(Result.Failure::class.java))
        assertThat("hasSuccessChanged is false", hasSuccessChanged, equalTo(false))
        assertThat("hasFailureChanged is true", hasFailureChanged, equalTo(true))
}
```